pipefunc/map/_storage_array/_zarr.py

```diff
@@ -60,24 +72,25 @@ def __init__(
             object_codec = CloudPickleCodec()
 
         chunks = select_by_mask(self.shape_mask, (1,) * len(self.shape), self.internal_shape)
-        self.array = zarr.open(
+        zarr_kwargs = {"mode": "a"} if _ZARR_MAJOR < 3 else {}  # noqa: PLR2004
+        self.array = create_array(
             self.store,
-            mode="a",
-            path="/array",
+            name="/array",
             shape=self.full_shape,
             dtype=object,
             object_codec=object_codec,
```

Comment:
Zarr v3 does not support object arrays yet. 
See https://zarr.readthedocs.io/en/stable/user-guide/arrays.html#missing-features-in-3-0

https://github.com/zarr-developers/zarr-python/issues/2617